yarn-project/noir-protocol-circuits/src/crates/types/src/abis/append_only_tree_snapshot.nr

```diff
@@ -1,12 +1,37 @@
 use dep::std::cmp::Eq;
+use crate::traits::Empty;
 
 struct AppendOnlyTreeSnapshot {
     root : Field,
     next_available_leaf_index : u32
 }
 
+global APPEND_ONLY_TREE_SNAPSHOT_LENGTH: Field = 2;
+
+impl AppendOnlyTreeSnapshot {
+    pub fn serialize(self) -> [Field; APPEND_ONLY_TREE_SNAPSHOT_LENGTH] {
+        [self.root, self.next_available_leaf_index as Field]
+    }
+
+    pub fn deserialize(serialized: [Field; APPEND_ONLY_TREE_SNAPSHOT_LENGTH]) -> AppendOnlyTreeSnapshot {
+        AppendOnlyTreeSnapshot {
+            root : serialized[0],
+            next_available_leaf_index : serialized[1] as u32
+        }
+    }
+}
+
 impl Eq for AppendOnlyTreeSnapshot {
     fn eq(self, other : AppendOnlyTreeSnapshot) -> bool {
         (self.root == other.root) & (self.next_available_leaf_index == other.next_available_leaf_index)
     }
 }
+
+impl Empty for AppendOnlyTreeSnapshot {
+    fn empty() -> Self {
```

Comment:
To be annoying, a tree can be empty without having root 0 🫠

Reply:
https://github.com/AztecProtocol/aztec-packages/pull/4154/commits/68cab0962ffdce5dca98952eae050d476181cfdf